ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -124,6 +124,11 @@ public static final class Headers {
      * has been modified after the value set for this header.
      */
     public static final String IF_MODIFIED_SINCE = "If-Modified-Since";
+    /**
+     * Header to be set by clients during a Put blob call to clearly specify that the client is aware of the
+     * UPDATE feature of named blob, and client does NOT wants to update when the named blob already exist.
+     */
+    public static final String NAMED_NOT_UPSERT = "x-ambry-named-not-upsert";
```

Comment:
We agreed to default to be upsert. Since the '`getBooleanHeader`' method was defaulting to 'false', so I used this 'NAMED_NOT_UPSERT' flag. I agree that double negative is confusing, so I just updated it to be positive flag, and also updated '`getBooleanHeader`' method to accept a 'defaultVal' (in the isUpsertForNamedBlob case, the defaultVal is true).